src/model/position.js

```diff
@@ -782,6 +818,41 @@ export default class Position {
 }
 
 /**
+ * Method used to expose root setter to child classes.
+ * @protected
+ * @param {module:engine/model/position~Position} position Position of which root should be modified.
+ * @param {module:engine/model/element~Element|module:engine/model/documentfragment~DocumentFragment} root Root of the position.
+ */
+export function setRoot( position, root ) {
+	position[ _root ] = root;
+}
+
+/**
+ * Method used to expose path setter to child classes.
+ * @protected
+ * @param {module:engine/model/position~Position} position Position of which path should be modified.
+ * @param {Array.<Number>} path Position path. See {@link module:engine/model/position~Position#path}.
+ */
+export function setPath( position, path ) {
```

Comment:
I would make this a protected class method.

Reply:
I have mixed feelings about this.

From one point we don't want to mess around with position path. On the other hand, though, we need this or `LivePosition` won't work (there is an exactly same problem with `LiveRange#start` and `LiveRange#end`).

If these are protected methods then we don't have to really refactor some parts of the code. Just use the new protected methods! But is this really the way to go?

So IDK, I am torn. I liked this as a utility function because it is more discouraging to use it. But IDK. Maybe `LiveRange` should be in the same file as `Range`. Then we would just use the symbol defined in `range.js`.

Reply:
For me, it the matter of convention. We have classes with protected methods exposed for other related classes in modules. For instance: https://github.com/ckeditor/ckeditor5-engine/blob/79da6fb1c0782cb6442527d3f2a50497b89b0985/src/model/operation/insertoperation.js#L84 We already had a talk about exposing protected methods for child classes at some point of designing editor class. We never did it as a separate export.

Reply:
I know what is the purpose of `@protected` tag and how we use it. The question is whether we want to have such `@protected` method at all. Because if we do, we can revert (or rather re-do) many of the changes proposed in this PR (at least in OT and tree walkers as these are used multiple time consecutively).

Reply:
Now I see that also there is a symbol used for the property. I would go with the `@protected` property (without a symbol) and public getters. We have such pair already in some places in the code. We can even write that the property is exposed only for `LiveRange` and should not be used anywhere.